src/parser.ts

```diff
@@ -365,31 +370,32 @@ export class Parser {
 
   pushPos() {
     if (this._posStack) {
-      this._posStack.push(this.getPos());
+      this._posStack.push(this.getPos() as Position);
```

Comment:
NOTE: TS has a postfix-`!` syntax as a simple cast to remove `null`/`undefined`, but just as with the `as` cast, both are type-level only with no runtime enforcement. 

This is also possibly unsound if somehow you end up with a defined `this._posStack` but an undefined `node.location`. Perhaps a better approach would be ensure the value is defined:

```suggestion
      const pos = this.getPos();
      if (pos) {
          this._posStack.push(pos);
      }
```

Reply:
Alternatively, you could change the body in this way (if you are using TS 3.7 or later):
```ts
pushPos() {
  const pos = this.getPos();
  if (pos) this._posStack?.push(pos);
}
```